### PR TITLE
ColaManhua (OhManhua): add Category and Char filters

### PR DESCRIPTION
**Pull request by he0119**
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/zh/onemanhua/src/eu/kanade/tachiyomi/extension/zh/onemanhua/Onemanhua.kt [hunk 1]
```diff
@@ -124,9 +124,12 @@ class Onemanhua : ConfigurableSource, ParsedHttpSource() {
     // Filter
     private class StatusFilter : Filter.TriState("已完结")
     private class SortFilter : Filter.Select<String>("排序", arrayOf("更新日", "收录日", "日点击", "月点击"), 2)
-
+    private class CategoryFilter : Filter.Select<String>("类型", arrayOf("全部", "热血", "玄幻", "恋爱", "冒险", "古风", "都市", "穿越", "奇幻", "其他", "少男", "搞笑", "战斗", "冒险热血", "重生", "爆笑", "逆袭", "后宫", "少年", "少女", "熱血", "系统", "动作", "校园", "冒險", "修真", "修仙", "剧情", "霸总", "大女主", "生活"), 0)
+    private class CharFilter : Filter.Select<String>("字母", arrayOf("全部", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"), 0)
```
Comment (beer-psi) [inline src/zh/onemanhua/src/eu/kanade/tachiyomi/extension/zh/onemanhua/Onemanhua.kt]: You can create a new `Filter.Select<String>` subclass to bind the display name to their respective category IDs:

```kotlin
open class UriPartFilter(displayName: String, private val vals: Array<Pair<String, String>>) :
    Filter.Select<String>(displayName, vals.map { it.first }.toTypedArray()) {
    fun toUriPart() = vals[state].second
}

private class CategoryFilter : UriPartFilter(
    "类型",
    arrayOf(
        "全部" to "",
        "热血" to "10023",
         // so on...
    )
)
```

And then, inside `searchMangaRequest`:
```kotlin
when (filter) {
   is CategoryFilter -> if (filter.state != 0) {
       url.addQueryParameter("mainCategoryId", filter.toUriPart())
   }
   // so on...
}
```

**Review by beer-psi — CHANGES_REQUESTED** [review]

**Comment by he0119** [comment]
It seems that order by create stop working now.

https://www.colamanga.com/show?orderBy=create

File: src/zh/onemanhua/build.gradle [hunk 2]
```diff
@@ -6,7 +6,7 @@ ext {
     extName = 'ColaManhua (OhManhua)'
```
Comment (stevenyomi) [inline src/zh/onemanhua/build.gradle]:
```suggestion
    extName = 'COLAMANGA'
```
Change the name field inside source class too.

File: src/zh/onemanhua/src/eu/kanade/tachiyomi/extension/zh/onemanhua/Onemanhua.kt [hunk 3]
```diff
@@ -122,11 +122,91 @@ class Onemanhua : ConfigurableSource, ParsedHttpSource() {
     override fun latestUpdatesFromElement(element: Element) = commonMangaFromElement(element)
 
     // Filter
+    open class UriPartFilter(displayName: String, private val vals: Array<Pair<String, String>>) :
+        Filter.Select<String>(displayName, vals.map { it.first }.toTypedArray()) {
+        fun toUriPart() = vals[state].second
+    }
     private class StatusFilter : Filter.TriState("已完结")
```
Comment (stevenyomi) [inline src/zh/onemanhua/src/eu/kanade/tachiyomi/extension/zh/onemanhua/Onemanhua.kt]: Refactor this into a string select filter (全部, 连载中, 已完结).

**Review by beer-psi — APPROVED** [review]

**Pull request merged**